--- a/libknet/links.c 
+++ b/libknet/links.c
@@ -896,7 +896,7 @@
 
 	if ((interval * 1000) < KNET_THREADS_TIMERES) {
 		log_warn(knet_h, KNET_SUB_LINK,
-			 "host: %u link: %u interval: %lu too small (%s). interval lower than thread_timer_res (%u ms) has no effect",
+			 "host: %u link: %u interval: %llu too small (%s). interval lower than thread_timer_res (%u ms) has no effect",
 			 host_id, link_id, interval, strerror(savederrno), (KNET_THREADS_TIMERES / 1000));
 	}
 
@@ -904,7 +904,7 @@
 		err = -1;
 		savederrno = EINVAL;
 		log_err(knet_h, KNET_SUB_LINK,
-			"host: %u link: %u pong timeout: %lu too small (%s). timeout cannot be less than thread_timer_res (%u ms)",
+			"host: %u link: %u pong timeout: %llu too small (%s). timeout cannot be less than thread_timer_res (%u ms)",
 			host_id, link_id, timeout, strerror(savederrno), (KNET_THREADS_TIMERES / 1000));
 		goto exit_unlock;
 	}